src/Microsoft.NetCore.Analyzers/CSharp/Performance/CSharpUsePropertyInsteadOfCountMethodWhenAvailable.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.NetCore.Analyzers.Performance;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Performance
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpUsePropertyInsteadOfCountMethodWhenAvailableAnalyzer
+        : UsePropertyInsteadOfCountMethodWhenAvailableAnalyzer
+    {
+        /// <summary>
+        /// Creates the operation actions handler.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <returns>The operation actions handler.</returns>
+        protected override OperationActionsHandler CreateOperationActionsHandler(OperationActionsContext context)
+            => new CSharpOperationActionsHandler(context);
+
+        /// <summary>
+        /// Handler for operaction actions for C#. This class cannot be inherited.
+        /// Implements the <see cref="Microsoft.NetCore.Analyzers.Performance.UsePropertyInsteadOfCountMethodWhenAvailableAnalyzer.OperationActionsHandler" />
+        /// </summary>
+        /// <seealso cref="Microsoft.NetCore.Analyzers.Performance.UsePropertyInsteadOfCountMethodWhenAvailableAnalyzer.OperationActionsHandler" />
+        private sealed class CSharpOperationActionsHandler : OperationActionsHandler
+        {
+            internal CSharpOperationActionsHandler(OperationActionsContext context)
+                : base(context)
+            {
+            }
+
+            protected override ITypeSymbol GetEnumerableCountInvocationTargetType(IInvocationOperation invocationOperation)
```

Comment:
I presume the language specific code is needed due to https://github.com/dotnet/roslyn/issues/23625 :(

Reply:
Yes!

Reply:
I think the logic here can be significantly signified as follows:

1. Base analyzer defines an abstract method `IOperation GetReceiverForExtensionMethodInvocation(IInvocationOperation`.
2. All the logic that you have in `GetEnumerableCountInvocationTargetType` overrides can be moved to the base type which uses the above helper to just find the IOperation on which to operation.
3. C# and VB language implementations should ideally then be just a single method override with a trivial one line implementation to address the core difference from https://github.com/dotnet/roslyn/issues/23625